src/value.c

```diff
@@ -571,36 +574,7 @@ int RSValue_Equal(const RSValue *v1, const RSValue *v2, QueryError *qerr) {
   v2 = RSValue_Dereference(v2);
 
   if (v1->t == v2->t) {
-    switch (v1->t) {
-      case RSValue_Number:
-        return cmp_numbers(v1, v2) == 0;
-        
-      case RSValue_String:
-        return cmp_strings(v1->strval.str, v2->strval.str, v1->strval.len, v2->strval.len) == 0;
-        
-      case RSValue_RedisString:
-      case RSValue_OwnRstring: {
-        size_t l1, l2;
-        const char *s1 = RedisModule_StringPtrLen(v1->rstrval, &l1);
-        const char *s2 = RedisModule_StringPtrLen(v2->rstrval, &l2);
-        return cmp_strings(s1, s2, l1, l2) == 0;
-      }
-      
-      case RSValue_Null:
-        return 1;
-
-      case RSValue_Array: {
-          if (v1->arrval.len != v2->arrval.len) return 0;
```

Comment:
This piece of code does not exist in the general comparison function, but we can indeed support array comparison in both cases (of equality and general comparison).

Reply:
Yeah, I figured it cannot hurt